backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -260,8 +261,13 @@ private Project insertProject(Project project, String workspaceName, UserInfo us
       deletedEntitiesQuery.setParameter("projectName", project.getName());
       List<String> deletedEntityIds = deletedEntitiesQuery.list();
       if (!deletedEntityIds.isEmpty()) {
-        roleService.deleteEntityResourcesWithServiceUser(
-            deletedEntityIds, ModelDBServiceResourceTypes.PROJECT);
+        try {
+          CommonUtils.registeredBackgroundUtilsCount();
+          roleService.deleteEntityResourcesWithServiceUser(
+              deletedEntityIds, ModelDBServiceResourceTypes.PROJECT);
+        } finally {
+          CommonUtils.unregisteredBackgroundUtilsCount();
+        }
```

Comment:
I don't understand how this solves a null pointer exception.  It appears to swallow the exception.  Please explain why an NPE is occurring.

Reply:
We have set the null value for the metadata at https://github.com/VertaAI/modeldb/blob/b32413a82f84c10c354517128f9d08e7ef6a30fa/backend/common/src/main/java/ai/verta/modeldb/common/authservice/RoleServiceUtils.java#L106 for the service user so it will broken at https://github.com/VertaAI/modeldb/blob/b32413a82f84c10c354517128f9d08e7ef6a30fa/backend/common/src/main/java/ai/verta/modeldb/common/authservice/AuthServiceChannel.java#L59 because metadata was null.

here added `CommonUtils.registeredBackgroundUtilsCount();` because whenever we are making call to other service from utils or any other places, we are increase the background util count to protect the API calls with empty headers.

So now you had question on mind, why we have used `CommonUtils.registeredBackgroundUtilsCount();`? right. So let say if we have just check the metadata null then there is possibility to user make a api call without metadata and that case MDB talking service user for that request. so to protect that scenario we had introduce that count mechanism.

Reply:
I agree with Cory. Even with the explanation, I don't know what is causing the NPE. What does `registeredBackgroundUtilsCount` do?

>So let say if we have just check the metadata null then there is possibility to user make a api call without metadata and that case MDB talking service user for that request. so to protect that scenario we had introduce that count mechanism.

This sounds like a hack instead of solving the actual problem, since you add an extra signal to infer what is the desired behavior indirectly. It also seems like we do that change regardless of whether the caller was the one to set the count or not. Seems like a global flag, which I believe I had reviewed and asked not to do at the time.

Reply:
We can improve and simplify this logic with new service and channel initialization which we have recently done but for now for the old design we should have to consider that background utils count logic.

Reply:
Ok. Let's make sure to remove any implicit signal between things as it makes bugs like these very, very easy =)

Reply:
Simply the bug was occurred as per following comments. forget about the `registeredBackgroundUtilsCount` as of now.

We have set the null value for the metadata at https://github.com/VertaAI/modeldb/blob/b32413a82f84c10c354517128f9d08e7ef6a30fa/backend/common/src/main/java/ai/verta/modeldb/common/authservice/RoleServiceUtils.java#L106 for the service user so it will broken at https://github.com/VertaAI/modeldb/blob/b32413a82f84c10c354517128f9d08e7ef6a30fa/backend/common/src/main/java/ai/verta/modeldb/common/authservice/AuthServiceChannel.java#L59 because metadata was null.